app/router.go

```diff
@@ -76,6 +77,11 @@ func Router(c collector) *mux.Router {
 	return router
 }
 
+func badRequest(r *http.Request, w http.ResponseWriter, err error) {
+	http.Error(w, err.Error(), http.StatusBadRequest)
+	log.Printf("Error procressing request for %s: %v", r.URL.Path, err)
+}
```

Comment:
[Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)):

- If you need it once, write it.
- If you need it twice, copy it.
- If you need it three times, factor it out.

You're only at step 2, ergo this function shouldn't exist yet :)